client/src/main/scala/com.crobox.clickhouse/stream/ClickhouseSink.scala

```diff
@@ -11,15 +11,30 @@ import scala.concurrent.duration._
 import scala.concurrent.{ExecutionContext, Future}
 
 case class ClickhouseIndexingException(msg: String, cause: Throwable, payload: Seq[String], table: String)
-    extends RuntimeException(msg, cause)
-case class Insert(table: String, jsonRow: String)
+  extends RuntimeException(msg, cause)
+
+sealed trait TableOperation {
+  def table: String
+
+  def jsonRow: String
+
+  def cluster: Option[String]
```

Comment:
Cluster should be removed from the interface. It should solely be put in the OptimizeOperation since it's only used there